.github/workflows/release-guardian.yaml

```
@@ -0,0 +1,126 @@
+on:
+  push:
+    tags:
+      - "release/guardian-app/*"  # Trigger on version tags for guardian app
+
+env:
+  NODE_VERSION: "22.12.0"
+  RUST_VERSION: "1.81.0"
+
+jobs:
+  build-and-release:
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - platform: macos-latest
+            arch: x64
+            target: x86_64-apple-darwin
+          - platform: macos-latest
+            arch: arm64
+            target: aarch64-apple-darwin
+          - platform: ubuntu-latest
+            arch: x64
+            target: x86_64-unknown-linux-gnu
+          - platform: windows-latest
+            arch: x64
+            target: x86_64-pc-windows-msvc
+
+    runs-on: ${{ matrix.platform }}
+    
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          submodules: recursive
+
+      - name: Extract tag name
+        id: get_tag
+        run: echo "::set-output name=tag::${GITHUB_REF#refs/tags/release/guardian-app/}"
```

[`::set-output` is deprecated](https://github.blog/changelog/2022-10-10-github-actions-deprecating-save-state-and-set-output-commands/) use:
```suggestion
        run: echo "tag=${GITHUB_REF#refs/tags/release/guardian-app/}"  >> $GITHUB_OUTPUT
```